### PR TITLE
Stable video diffusion

### PR DESCRIPTION
**Pull request by Fox4Song**
Backend service for Stable Video Diffusion

### CODE REVIEW

File: server/vid.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+@svc.api(input=Image(), output=Image())
+def img2vid(input_data):
+    res = svd_runner.run(**input_data)
+    return res[0][0]
+
+app = FastAPI(root_path="/v9")
+svc.mount_asgi_app(app)
+
+#@app.get("/status")
```
Comment (acowusu) [inline server/vid.py:35]: Include status endpoint for observability
```suggestion
@app.get("/status")
```

File: server/vid.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#async def status():
```
Comment (acowusu) [inline server/vid.py:36]:
```suggestion
async def status():
```

File: server/vid.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#    return {"status": "ok"}
```
Comment (acowusu) [inline server/vid.py:37]:
```suggestion
    return {"status": "ok"}
```

File: server/vid.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+@app.post("/video")
+async def generate_video(image_file: UploadFile = Form(...), fps: Annotated[int, Form()]=7, video_length: Annotated[int, Form()]=10):
+    data = await image_file.read()
+    image = PIL.Image.open(BytesIO(data))
+    image = image.resize((1024, 576))
+    image = PIL.ImageOps.exif_transpose(image)
+    image = image.convert("RGB")
+    #image = load_image(image_file)
```
Comment (acowusu) [inline server/vid.py:46]:
```suggestion
```

File: desktop/electron/videoGen.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { downloadFile } from "./reportProcessing";
+import { getProjectPath } from "./metadata";
+import fs from "fs";
```
Comment (acowusu) [inline desktop/electron/videoGen.ts:3]:
```suggestion
import { readFile } from "node:fs/promises";
```

File: desktop/electron/videoGen.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+    // const filePromise = new Promise((resolve, reject) => {
+    //     fs.readFile(imagePath, (err, data) => {
+    //       if (err) { reject(err); }
+    //       resolve(dataurl.convert({ data, mimetype: 'image/png' }));
+    //     });
+    // });
+    // const imageUrl = await filePromise as string;
+
+    const imageBuffer = fs.readFileSync(imagePath);
```
Comment (acowusu) [inline desktop/electron/videoGen.ts:20]: Async alternative preferred as non blocking
```suggestion
    const imageBuffer = await readFile(imagePath);
```

File: desktop/electron/videoGen.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    console.log("Image Path: ", imagePath);
+    const form = new FormData();
+
+    // const filePromise = new Promise((resolve, reject) => {
+    //     fs.readFile(imagePath, (err, data) => {
+    //       if (err) { reject(err); }
+    //       resolve(dataurl.convert({ data, mimetype: 'image/png' }));
+    //     });
+    // });
+    // const imageUrl = await filePromise as string;
```
Comment (acowusu) [inline desktop/electron/videoGen.ts:18]: Remove commented out code
```suggestion
```

File: desktop/electron/videoGen.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+    const imageBuffer = fs.readFileSync(imagePath);
+    const imageBlob = new Blob([imageBuffer], { type: 'image/png' });
```
Comment (acowusu) [inline desktop/electron/videoGen.ts:21]: Is there  a reason for using PNG here, the images generated from Open Journey are JPEGS

**Review by acowusu — CHANGES_REQUESTED** [review]
Looks good, Given a few suggestions. 

In addition can you make a few changes so that we can turn on and off the service, and keep track of whether it is running:
*  commit the service file 
* Add the endpoint in `status.ts` (something like this)
 ```    {
        url: "https://iguana.alexo.uk/v7/status",
        name: "Sound Effects API",
    }
```
*  Add a record of the service in `server/experimental/control.py`
```
allowed_services = [
    "fastapi-img",
    "fastapi-tts",
    "fastapi-llm",
    "fastapi-music",
    "fastapi-audiofx",
   "YOUR SERVICE NAME"
]
mapping = {
    "Image API": "fastapi-img",
    "Text to speech API": "fastapi-tts",
    "LLM API": "fastapi-llm",
    "Music API": "fastapi-music",
    "Sound Effects API": "fastapi-audiofx"
  "YOUR SERVICE DISPLAY NAME" "YOUR SERVICE NAME"

}

```

File: desktop/electron/videoGen.ts (hunk 8 above)
Comment (Fox4Song) [reply desktop/electron/videoGen.ts:14]: Sorry was just using png to test local images. I can change it to so that we can accept any image mime type.

**Pull request closed without merging**